metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -185,34 +234,49 @@ def extract_lineage(
         )
 
         logger.debug(
-            f"PowerBI virtual table {table.full_name} and it's upstream dataplatform tables = {upstream_dpts}"
+            f"PowerBI virtual table {table.full_name} and it's upstream dataplatform tables = {upstream_lineage}"
         )
 
-        for upstream_dpt in upstream_dpts:
-            if (
-                upstream_dpt.data_platform_pair.powerbi_data_platform_name
-                not in self.__config.dataset_type_mapping.keys()
-            ):
-                logger.debug(
-                    f"Skipping upstream table for {ds_urn}. The platform {upstream_dpt.data_platform_pair.powerbi_data_platform_name} is not part of dataset_type_mapping",
+        for lineage in upstream_lineage:
+            for upstream_dpt in lineage.upstreams:
+                if (
+                    upstream_dpt.data_platform_pair.powerbi_data_platform_name
+                    not in self.__config.dataset_type_mapping.keys()
+                ):
+                    logger.debug(
+                        f"Skipping upstream table for {ds_urn}. The platform {upstream_dpt.data_platform_pair.powerbi_data_platform_name} is not part of dataset_type_mapping",
+                    )
+                    continue
+
+                upstream_table_class = UpstreamClass(
+                    upstream_dpt.urn,
+                    DatasetLineageTypeClass.TRANSFORMED,
                 )
-                continue
 
-            upstream_table_class = UpstreamClass(
-                upstream_dpt.urn,
-                DatasetLineageTypeClass.TRANSFORMED,
-            )
+                upstream.append(upstream_table_class)
 
-            upstream.append(upstream_table_class)
+                # Add column level lineage if any
+                cll_lineage.extend(
+                    self.make_fine_grained_lineage_class(
+                        lineage=lineage,
+                        dataset_urn=ds_urn,
+                    )
+                )
 
         if len(upstream) > 0:
-            upstream_lineage = UpstreamLineageClass(upstreams=upstream)
+
+            upstream_lineage_class: UpstreamLineageClass = UpstreamLineageClass(
+                upstreams=upstream,
+                fineGrainedLineages=cll_lineage if len(cll_lineage) > 0 else None,
```

Comment:
Nice

Reply:
We can actually make this even simpler - because `bool([])` is false, something like this will work

```suggestion
                fineGrainedLineages=cll_lineage or None,
```

Reply:
done